### ISSUE TITLE
[a11y]: DataTable [dynamic] - settings button menu extraneous announcement with SR

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14266]
### Package

@carbon/react

### Browser

FF

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description
JAWS extraneous announcement after selecting "Add Row"

After selecting the Add row option from the settings menu, JAWS announces the entire content from the newly added Row. This announcement is unnecessary and can be distracting to the user

### Recommended Fix
remove the extraneous announcement

Any redundant, excessively wordy and extraneous announcement can be distracting for the screen reader users. 

### WCAG 2.1 Violation

4.1.2 Name, Role, and Value

### Reproduction/example
Record `Number 69` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-dynamic--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
Checked with our accessibility team, and this issue still exists.
